cpp/src/parquet/encoding.cc

```diff
@@ -2937,11 +2937,16 @@ ByteStreamSplitDecoder<DType>::ByteStreamSplitDecoder(const ColumnDescriptor* de
 template <typename DType>
 void ByteStreamSplitDecoder<DType>::SetData(int num_values, const uint8_t* data,
                                             int len) {
-  DecoderImpl::SetData(num_values, data, len);
-  if (num_values * static_cast<int64_t>(sizeof(T)) > len) {
-    throw ParquetException("Data size too small for number of values (corrupted file?)");
+  if (num_values * static_cast<int64_t>(sizeof(T)) < len) {
+    throw ParquetException("Data size too large for number of values (padding file?)");
```

Comment:
It might be nice to be more specific.
```suggestion
    throw ParquetException("Data size too large for number of values (padding in byte stream split data page?)");
```